src/webgpu/shader/execution/robust_access_vertex.spec.ts

```diff
@@ -0,0 +1,40 @@
+export const description = `
+Test vertex attributes behave correctly (no crash / data leak) when accessed out of bounds
+`;
+
+/*
+Test coverage:
+
+The following will be parameterized (all combinations tested):
+
+1) Draw call indexed? (false / true)
+  - Run the draw call using an index buffer
+
+2) Draw call indirect? (false / true)
+  - Run the draw call using an indirect buffer
+
+3) Draw call parameter (vertexCount, firstVertex, indexCount, firstIndex, baseVertex, instanceCount,
+  firstInstance)
+  - The parameter which will go out of bounds. Filtered depending on if the draw call is indexed.
```

Comment:
tip: perhaps nice to write this as

```typescript
.combine(pbool('indexed'))
.expand(function* (p) {
  if (p.indexed) {
    yield* ['indexCount', 'firstIndex', 'baseVertex'];
  } else {
    yield* ['vertexCount', 'firstVertex'];
  }
  yield* ['instanceCount', 'firstInstance'];
})
```

Reply:
Nice. My first attempt was using two enums, something like NonIndexedDrawCallParameter and IndexedDrawCallParameter, and then doing:
```typescript
.combine(pbool('indexed'))
.combine(poptions('drawCallTestParameter', kDrawCallParameters))
.filter(({ indexed, drawCallTestParameter }) =>
    indexed && drawCallTestParameter in IndexedDrawCallParameter ||
    !indexed && drawCallTestParameter in NonIndexedDrawCallParameter)
```

Do you prefer one or the other?

Reply:
Either is fine, whatever works well.
Realized my suggestion was not right because it doesn't have a key, but you can fix that.

Maybe this would be even simpler:
```ts
.combine(pbool('indexed'))
.expand(p => poptions('drawCallTestParameter', p.indexed ?
  ['indexCount', 'instanceCount', 'firstIndex', 'baseVertex', 'firstInstance'] :
  ['vertexCount', 'instanceCount', 'firstVertex', 'firstInstance']))
```

Or depending on how you're actually writing the test:

```ts
function makeSequence(n) { return [... Array(n).keys()]; }

...
.combine(pbool('indexed'))
.expand(p => poptions('drawCallTestParameter', makeSequence(p.indexed ? 5 : 4))

...

const args = p.indexed ? [1, 1, 1, 1, 1] : [1, 1, 1, 1];
args[p.drawCallTestParameter] = 100;
```

or if you want more explicitness/control over the args:

```ts
.combine(pbool('indexed'))
.expand(p => poptions('drawCallArgs', p.indexed ? [
  [100,1,1,1,1],
  [1,100,1,1,1],
  ...
] : [
  [100,1,1,1],
  [1,100,1,1],
  ...
])
```

Reply:
Thank you very much! Will try these.